core/src/main/java/org/apache/carbondata/core/reader/CarbonFooterReaderV3.java

```diff
@@ -49,7 +57,24 @@ public CarbonFooterReaderV3(String filePath, long offset) {
   public FileFooter3 readFooterVersion3() throws IOException {
     ThriftReader thriftReader = openThriftReader(filePath);
     thriftReader.open();
-    //Set the offset from where it should read
+
+    // If footer offset is 0, means caller does not set it,
+    // so we read it from the end of the file
+    if (footerOffset == 0) {
+      Configuration conf = FileFactory.getConfiguration();
+      FileFactory.FileType fileType = FileFactory.getFileType(filePath);
+      DataInputStream reader = FileFactory.getDataInputStream(filePath, fileType, conf);
+      long skipBytes = fileSize - CarbonCommonConstants.LONG_SIZE_IN_BYTE;
+      long skipped = reader.skip(skipBytes);
+      if (skipped != skipBytes) {
+        throw new IOException(String.format(
+            "expect skip %d bytes, but actually skipped %d bytes", skipBytes, skipped));
+      }
+      footerOffset = reader.readLong();
+      reader.close();
```

Comment:
The reader will not be closed when the the previous code throw any exception.

Reply:
fixed, using try with resource